x-pack/plugins/stack_alerts/public/rule_types/es_query/expression/es_query_expression.tsx

```diff
@@ -249,9 +263,10 @@ export const EsQueryExpression: React.FC<
           width="100%"
           height="200px"
           value={xJson}
-          onChange={(xjson: string) => {
+          onChange={async (xjson: string) => {
             setXJson(xjson);
             setParam('esQuery', convertToJson(xjson));
+            await refreshEsFields();
```

Comment:
`refreshEsFields` issues an HTTP request to the server so calling it on each `onChange` means we're hitting the server with every keystroke in the DSL editor. Since we're parsing out the runtime fields from the JSON itself, it doesn't seem like we should be adding these additional network calls.

Reply:
That makes sense, I'll remove it

Reply:
Actually I think we need to to trigger refreshing the fields, other wise the fields will only update when an index is selected. What do you think about using debounce to reduce the number of network calls?

Reply:
I think it's fine not to get the latest state of the fields if they're changed while some is in the UI creating a rule. Right now, we're parsing out the runtime fields, passing them into `refreshEsFields`, making the HTTP call and appending the runtime fields to the value that's returned. Could we just keep the existing logic for `refreshEsFields` and parse the runtime fields `onChange` and add them to the fields returned in the initial `refreshEsFields` call?

Reply:
Sure, we can do that instead!

Reply:
Resolved in this commit https://github.com/elastic/kibana/pull/160319/commits/1167e9ad7b0ecac74eb191e0604f1b971c85ac50